lib/dgram.js

```diff
@@ -272,6 +267,65 @@ Socket.prototype.send = function(buffer,
   if (offset + length > buffer.length)
     throw new RangeError('Offset + length beyond buffer length');
 
+  return buffer.slice(offset, offset + length);
+}
+
+function fixBuffer(buffer) {
+  var length = 0;
+
+  for (var i = 0, l = buffer.length; i < l; i++) {
+    if (typeof buffer[i] === 'string')
+      buffer[i] = new Buffer(buffer[i]);
+    else if (!(buffer[i] instanceof Buffer))
+      return -1;
+
+    length += buffer[i].length;
+  }
+
+  return length;
+}
+
+function enqueue(self, toEnqueue) {
+  // If the send queue hasn't been initialized yet, do it, and install an
+  // event handler that flushes the send queue after binding is done.
+  if (!self._sendQueue) {
+    self._sendQueue = [];
+    self.once('listening', function() {
+      // Flush the send queue.
+      for (var i = 0; i < self._sendQueue.length; i++)
+        self.send.apply(self, self._sendQueue[i]);
+      self._sendQueue = undefined;
+    });
```

Comment:
The only variable I see here that's being used is `self`. Since `'once'` is called on the `self` object it will also be the `this` in the callback. So this function can be hoisted.